test/scripts-test.js

```diff
@@ -1,7 +1,194 @@
 import { expect } from 'chai';
 
 //FUNCTIONS FROM THE SCRIPT DEFINITIONS FILE
-const { generateRandomUserID, findUserByEmail, addDataToCurrentUser, findStepGoalAverage, getHydrationFor7Days, calculateTotalHydration, findDistanceTraveled, ouncesPerDay, calculateAverageHoursSlept, calculateAverageSleepQuality, hoursSleptGivenDate, sleepQualityGivenDate, getSleepFor7Days, getSleepQualityFor7Days, checkStepGoal, minutesActiveGivenDate, checkStepGoal7Days, numberOfStepsGivenDate, findBottomDrinkers, findBottomSleepers, findBottomStepTakers } = require('../src/scriptDefinitions');
+const { 
+    addDataToCurrentTraveler,
+    getImageURLsOfPendingTrips,
+    getImageURLsOfPastTrips,
+    getImageURLsOfFutureTrips,
+	calculateTotalSpentOnTrips,
+	calculateTripCost
+    } = require('../src/scriptDefinitions');
 
 //ACCESS TO THE TEST DATA
-const { users, hydrationData, activityData, sleepData } = require('../src/data/testData');
\ No newline at end of file
+const { 
+    travelers,
+    trips,
+    destinations
+    } = require('../src/data/testData');
+
+
+/////////TESTS TO CREATE THE TRAVELER/////////
+    describe('addDataToCurrentTraveler', function () {
+      it('should return null if travelerNumber is not found in travelersData', function () {
+        const result = addDataToCurrentTraveler(999, travelers, trips, destinations);
+        expect(result).to.be.null;
+      });
+    
+      it('should return the complete current traveler object when travelerNumber is found', function () {
+        const travelerNumber = 1; 
+        const result = addDataToCurrentTraveler(travelerNumber, travelers, trips, destinations);
+    
+        const expectedTraveler = travelers.find((traveler) => traveler.id === travelerNumber);
+    
+        const expectedTrips = trips.filter((trip) => trip.userID === travelerNumber);
+    
+        const expectedDestinations = expectedTrips.map((trip) =>
+          destinations.find((destination) => destination.id === trip.destinationID)
+        );
+    
+        const expected = {
+          traveler: expectedTraveler,
+          trips: expectedTrips,
+          destinations: expectedDestinations,
+        };
+    
+        expect(result).to.deep.equal(expected);
+      });
```

Comment:
@butlertree I don't think you need to do this much heavy lifting within your tests themselves. The actual function should do all the work, so all you really need to do is check that invoking the function with certain data gives you the result you're looking for. 

Here's an example from my test suite where I'm testing a very similar function called getCurrentTraveler:
```it('should add trips and destination data to user object', function () {
    expect(getCurrentTraveler(travelers[0], trips, destinations)).to.deep.equal({
      traveler: { id: 1, name: 'Ham Leadbeater', travelerType: 'relaxer' },
      trips: [
        {
          id: 117,
          userID: 1,
          destinationID: 28,
          travelers: 3,
          date: '2021/01/09',
          duration: 15,
          status: 'approved',
          suggestedActivities: []
        }
      ],
      destinations: [
        {
          id: 28,
          destination: 'San Juan, Puerto Rico',
          estimatedLodgingCostPerDay: 70,
          estimatedFlightCostPerPerson: 900,
          image: 'https://images.unsplash.com/photo-1580237541049-2d715a09486e?ixlib=rb-1.2.1&ixid=eyJhcHBfaWQiOjEyMDd9&auto=format&fit=crop&w=2090&q=80',
          alt: 'white and brown concrete buildings near sea under white clouds during daytime'
        }
      ]
    });
```

All I'm doing there in the `expect` is invoking the function using specific arguments, then showing what I actually expect to see in the return of the function. For my test data, I have a user that only has one trip and one destination, a user that has multiple trips and destinations, and a user that has no trips or destinations. That way, I can test a couple happy paths, as well as sad paths. 

If you update your tests that way, it appears cleaner and you can be more certain that the tests are actually testing your function directly, if that makes sense.  Not sure if that all makes sense written down, but I'm more than happy to explain further on a call, if you like!  🙂